Improve code readability

juanmbellini opened the pull request:
Closes #30

juanmbellini committed:
Improve code readibility of UserController

- Add a method in JerseyControllerHelper to avoid calling another method with empty map
- Improve Javadoc of JerseyControllerHelper

juanmbellini committed:
Improve code readibility of ReviewController
- Add some TODOs
- Change some path params (were not documented) to be consistent with the rest of the code

juanmbellini committed:
Improve code readibility of GameController

lipusal commented on paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/controller/ReviewJerseyController.java:
```diff
@@ -138,9 +140,9 @@ public Response deleteReview(@PathParam("id") final long reviewId) {
-                .header("Access-Control-Allow-Methods", "PUT,DELETE,OPTIONS,POST,GET")
-                .header("Access-Control-Allow-Headers", "Content-Type");  //Required by CORS
+                .type(MediaType.TEXT_HTML)
+                .header("Access-Control-Allow-Methods", "PUT,DELETE,OPTIONS,POST,GET") // TODO: PUT and DELETE are not supported methods for /reviews
```
Por qué no? Deberíamos permitirlos, las reviews tienen los botones de edit y delete

juanmbellini replied:
Sisi, pero el endpoint para eso es ```/reviews/:id```, no ```/reviews```

lipusal commented on paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/controller/ReviewJerseyController.java:
```diff
@@ -152,57 +154,52 @@ public Response reviewOptions() {
+    @Path("/{id : \\d+}/likes")
+    public Response likeReview(@SuppressWarnings("RSReferenceInspection") @PathParam("id") final long reviewId) {
+        JerseyControllerHelper.checkParameters(JerseyControllerHelper.getParametersWrapper()
+                .addParameter("id", reviewId, id -> id <= 0));
```
Me parece más claro el código que estaba antes, y de usar `reviewId`, aunque el segundo no es tan importante

juanmbellini replied:
Claro, pero al documentación dice ```/reviews/:id``` para el resto de los endpoints. Sino tengo que cambiar todos los demás.
Además, por otro lado, estamos dentro de "reviews" por lo que, teniendo solo ":id", se sobreentiende que es el "id del review".

lipusal replied:
Entiendo lo de `id`, me parecía poco claro la manera nueva de chequear errores. Claramente ya es demasiado tarde para que lo reviertas así que ya fue

juanmbellini replied:
👍

lipusal commented on paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/controller/UserJerseyController.java:
```diff
@@ -116,9 +124,9 @@ public Response getById(@PathParam("id") final long id) {
-        }
+        JerseyControllerHelper.checkParameters(JerseyControllerHelper.getParametersWrapper()
+                .addParameter("username", username, Objects::isNull));
+
```
Me parece extremadamente verborrágico y al pedo, lo anterior era más conciso e iba al grano (chequear que algo no sea null)

juanmbellini replied:
Lo hice para mantener consistencia. El método ```JerseyControllerHelper#checkParameters``` sirve para verificar parámetros todos juntos (en caso de que te equivocaste en más de uno, te avisa por todos). Si uso de la manera de antes, te puedo avisar de a uno.
Además, de esta manera, centralizo la lógica de cómo resolver este problema (qué hago y cómo lo hago, y cómo le doy aviso al cliente).

lipusal replied:
Por lo de centralizar el comportamiento te lo permito, pero sino preferiría el anterior porque de todas maneras acá estás chequeando un solo valor. Está bien lo de centralizar comportamiento igual, pero quedan re anti-intuitiva estas líneas (hay un `Objects::isNull` en vez de `Objects::isNotNull`, a rápida vista parece que esto está chequeando que el parámetro **sí** sea NULL). No me sorprendería que recibamos críticas por esto, pero dejalo

juanmbellini replied:
👍

lipusal commented on paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/controller/UserJerseyController.java:
```diff
@@ -301,19 +307,19 @@ public Response getFeedThreads(@SuppressWarnings("RSReferenceInspection") @PathP
-        }
+        JerseyControllerHelper.checkParameters(JerseyControllerHelper
+                .getPaginationReadyParametersWrapper(pageSize, pageNumber)
+                .addParameter("id", userId, id -> id <= 0));
```
Ídem acá y en todos los demás lugares

juanmbellini replied:
En este caso, estoy chequeando los parámetros ```pageSize``` y ```pageNumber```, además de ```id```.

lipusal commented on paw-webapp/webapp/src/main/java/ar/edu/itba/paw/webapp/controller/UserJerseyController.java:
```diff
@@ -415,13 +414,12 @@ public Response addPlayStatus(@PathParam("id") final long userId,
-        }
+        JerseyControllerHelper.checkParameters(JerseyControllerHelper.getParametersWrapper()
+                .addParameter("id", userId, id -> id <= 0)
+                .addParameter("gameId", gameId, id -> id <= 0));
```
Esto sí me parece piola

juanmbellini replied:
👍

lipusal reviewed (changes requested):
Algunos comments generales pero otros que sí me parece que habría que revisar

juanmbellini committed:
Improve code readibility of ShelfController

juanmbellini committed:
Improve code readibility of ThreadController

juanmbellini committed:
Split stream operations in UserDetailsServiceImpl

juanmbellini committed:
Refactor some DTOs

- Change stream().collect(Collectors.toList()) into new ArrayList<>(game.getPictureUrls())
- Change a ternary initialization (which checked null) into the Optional approach

juanmbellini committed:
Change UserServiceImpl#getWithSocialStuff method

- Change ternary approach to Optional#filter approach

juanmbellini committed:
Change ReviewServiceImpl#getReviews
- Split some lines to be more readable

juanmbellini committed:
Change UserServiceImpl#recommendGames
- Replace null comparison with method reference

juanmbellini committed:
Fix DaoHelper#findByFields method
- Change params.entrySet().forEach(entry -> query.setParameter()) into params.forEach(query::setParameter)
